doc/source/whatsnew/v0.21.0.txt

```diff
@@ -358,6 +358,7 @@ Indexing
 - Allow unicode empty strings as placeholders in multilevel columns in Python 2 (:issue:`17099`)
 - Bug in ``.iloc`` when used with inplace addition or assignment and an int indexer on a ``MultiIndex`` causing the wrong indexes to be read from and written to (:issue:`17148`)
 - Bug in ``.isin()`` in which checking membership in empty ``Series`` objects raised an error (:issue:`16991`)
+- When a numeric ``Index``'s :func:`get_indexer` is passed a boolean target it now returns the correct indexer  (:issue:`16877`)
```

Comment:
@jreback @jorisvandenbossche : I don't quite remember the formatting for `:func:` and stuff, but somehow that beginning part of the `whatsnew` looks incorrectly formatted.  However, correct me if I'm wrong.

Reply:
I would actually re-write this message so it is more user-oriented (the typical user does not know Index.get_indexer). In the end it is about boolean values that now raise when indexing a numeric index or dropping from such an index?

@gfyoung for `:func:`, to have it working, in this case you would need to write `` :func:`Index.get_indexer` `` (but as I said, I wouldn't add that, as typical user does not need to use that function)

Reply:
@jorisvandenbossche : Ah, gotcha, thanks for clarifying!

Reply:
agree, this issue is all about passing a boolean indexer to ``DataFrame.drop()``, so say that.